examples/scene_test.rs

```diff
@@ -1,182 +1,125 @@
+use ggez::Context;
 use ggez::GameResult;
-use std::time::Duration;
 
 use ggez_goodies::scene::*;
 
+struct SharedState {}
+
 struct MainState {
-    font: graphics::Font,
-    message_text: graphics::Text,
+    scenes: SceneStack<SharedState>,
 }
 
-/// A bootstrap scene whose only purpose is to
-/// load all the other bits necessary...
-struct StartScene;
-impl SavedScene<MainState> for StartScene {
-    fn load(&self) -> Box<Scene<MainState>> {
-        Box::new(StartScene)
-    }
-    fn name(&self) -> &str {
-        "Starting Scene"
+impl MainState {
+    fn new(ctx: &mut Context) -> GameResult<MainState> {
+        let mut scenes = SceneStack::new(ctx, SharedState {});
+        scenes.switch(SceneSwitch::push(StartScene { switch: false }));
+        Ok(MainState { scenes })
     }
 }
 
-impl Scene<MainState> for StartScene {
-    fn unload(&mut self) -> Box<SavedScene<MainState>> {
-        Box::new(StartScene)
-    }
-
-
-    fn update(&mut self,
-              _ctx: &mut ggez::Context,
-              _dt: Duration,
-              state: &mut SceneStore<MainState>)
-              -> GameResult<Option<String>> {
-        let s1 = SavedScene1::new("Scene 1", "Scene 2");
-        let s2 = SavedScene1::new("Scene 2", "Scene 1");
-        state.add(s1);
-        state.add(s2);
-        Ok(Some("Scene 1".to_string()))
-    }
-
-    fn draw(&mut self,
-            _ctx: &mut ggez::Context,
-            _store: &mut SceneStore<MainState>)
-            -> GameResult<()> {
+impl event::EventHandler for MainState {
+    fn update(&mut self, ctx: &mut Context) -> GameResult {
+        self.scenes.update(ctx);
+        self.scenes.input(SceneEvents::None, ctx, true);
         Ok(())
     }
-}
 
-impl GameData<MainState> for MainState {
-    fn load(ctx: &mut ggez::Context, _conf: &conf::Conf) -> GameResult<Self> {
-        let font = graphics::Font::new(ctx, "DejaVuSerif.ttf", 16)?;
-
-        let text = graphics::Text::new(ctx, "Press space to switch to the next scene.", &font)?;
-        Ok(MainState {
-            font: font,
-            message_text: text,
-        })
-    }
-    fn starting_scene() -> Box<SavedScene<MainState>> {
-        Box::new(StartScene)
+    fn draw(&mut self, ctx: &mut Context) -> GameResult {
+        self.scenes.draw(ctx);
+        Ok(())
     }
 }
 
-#[derive(Clone, Debug)]
-struct SavedScene1 {
-    time_unloaded: f64,
-    name: String,
-    next_scene: String,
+struct StartScene {
+    switch: bool,
 }
 
-impl SavedScene1 {
-    fn new(name: &str, next_scene: &str) -> Self {
-        SavedScene1 {
-            time_unloaded: 0.0,
-            name: name.to_string(),
-            next_scene: next_scene.to_string(),
+impl Scene<SharedState> for StartScene {
+    fn update(
+        &mut self,
+        _: &mut SharedState,
+        ctx: &mut ggez::Context,
+    ) -> SceneSwitch<SharedState, SceneEvents, ggez::Context> {
+        if ctx.keyboard.is_key_just_pressed(VirtualKeyCode::K) {
+            println!("Hello");
```

Comment:
This should probably be removed.

Also, the previous version used "space" to switch scenes and printed that in the display, but it's not a big deal.

Reply:
What do you mean? The println!() or the whole scene i'm a little confused

Reply:
Do you mean the println!() or something else?

Reply:
Cleaned up the SceneSwitch generics and removed the println